pkg/prometheus/agent/statefulset_test.go

```diff
@@ -52,75 +52,29 @@ func TestListenTLS(t *testing.T) {
 }
 
 func TestWALCompression(t *testing.T) {
-	tests := []struct {
-		version       string
-		enabled       *bool
-		expectedArg   string
-		shouldContain bool
-	}{
-		// Nil should not have either flag.
-		{"v2.30.0", ptr.To(false), "--storage.agent.wal-compression", false},
-		{"v2.32.0", nil, "--storage.agent.wal-compression", false},
-		{"v2.32.0", ptr.To(false), "--no-storage.agent.wal-compression", true},
-		{"v2.32.0", ptr.To(true), "--storage.agent.wal-compression", true},
-	}
+	testcases := createTestCasesForTestWALCompression()
 
-	for _, test := range tests {
-		sset, err := makeStatefulSetFromPrometheus(monitoringv1alpha1.PrometheusAgent{
-			Spec: monitoringv1alpha1.PrometheusAgentSpec{
-				CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
-					Version:        test.version,
-					WALCompression: test.enabled,
-				},
-			},
-		})
+	for _, test := range testcases {
+		sset, err := makeStatefulSetFromPrometheus(
+			makePrometheusAgentForTestWALCompression(
+				test.version,
+				test.enabled))
 		require.NoError(t, err)
 
-		promArgs := sset.Spec.Template.Spec.Containers[0].Args
-		found := false
-		for _, flag := range promArgs {
-			if flag == test.expectedArg {
-				found = true
-				break
-			}
-		}
-
-		if found != test.shouldContain {
-			if test.shouldContain {
-				t.Fatalf("expected Prometheus args to contain %v, but got %v", test.expectedArg, promArgs)
-			} else {
-				t.Fatalf("expected Prometheus args to NOT contain %v, but got %v", test.expectedArg, promArgs)
-			}
-		}
```

Comment:
The change you've made about this looks pretty good!
 
```go
require.Equal(t, shouldContain, found)
```
Looks a lot better than the nested ifs!

I'm just not sure about abstracting this into a separate function, for the same reasons above.